src/renderers/gl/renderer.cpp

```diff
@@ -379,32 +379,30 @@ void mrg::Renderer::draw(mg::Renderable const& renderable) const
 {
     auto const texture = gl_interface->as_texture(renderable.buffer());
     auto const clip_area = renderable.clip_area();
+    glm::mat4 transform = renderable.transformation();
+    if (texture->layout() == mg::gl::Texture::Layout::TopRowFirst)
+    {
+        // GL textures have (0,0) at bottom-left rather than top-left
+        // We have to invert this texture to get it the way up GL expects.
+        transform *= glm::mat4{
+            1.0, 0.0, 0.0, 0.0,
+            0.0, -1.0, 0.0, 0.0,
+            0.0, 0.0, 1.0, 0.0,
+            0.0, 0.0, 0.0, 1.0
+        };
```

Comment:
Drive by question: do we really want to invert before the transform? And not after?

I've not read the surrounding code to check, just been triggered because the snippet looks like #3188

Reply:
I fixed this PR so that we no longer use it, so this would be unrelated. Buuuut, I wonder as well! This `transform` is actually the `model` matrix for the buffer. I _imagine_ that we would want to invert _and then_ apply this transformation.

For example: If we rotate the buffer by 45 degrees, we will want to rotate the _inverted buffer_ instead of the pre-inverted buffer. In the latter case, we would have a resulting buffer that is rotated 135 degrees, which would be incorrect.

Reply:
This is probably a fix for another pull request though. And I would at least have to confirm my thoughts before throwing it in...